app/helpers/visit_helper.rb

```diff
@@ -18,44 +16,45 @@ def prison_name
   end
 
   def prison_names
-    Rails.configuration.prison_data.keys.sort
+    Prison.names
   end
 
   def prison_data(source=visit)
-    Rails.configuration.prison_data.fetch(source.prisoner.prison_name.to_s)
+    prison_name = source.is_a?(String) ? source : source.prisoner.prison_name.to_s
```

Comment:
A couple of things stand out here - whether they are addressed in this PR or not is a different matter.

Firstly, minor niggle, we are no longer returning a hash, so calling it data I feel takes away from it being an object in its own right now & is the start of it reading a little inconsistent.. sometimes we have prisons sometimes we have prison_data objects.

I also seem to recall seeing a number of similar methods that are auto seeded, and can take different inputs (which in most cases is a lazy way of defining a common interface) in this case `String` or `Visit`.

I think we can clear this up a little.. We can fall into two camps here either always pass the object to the method or accept that it is always available to us - `prison_data` seems to be commonly called without arguments so to make it consistent we can go with the latter & do away with having this method take any argument as `visit` is a global helper method defined in `ApplicationController` & is always available.

Doing a grep for this method I can find only 2 instances of it being called with args.

```bash
$ git grep "prison_data(" | grep -v def

app/helpers/visit_helper.rb:    slug = prison_data(visit).finder_slug || visit.prisoner.prison_name.parameterize
app/helpers/visit_helper.rb:    nomis_id = prison_data(prison_name).nomis_id
```

For the `slug` assignment we don't need to pass anything as its auto seeded with the visit anyway/as mentioned earlier visit is always available within the scope of this method.

For the `nomis_id` assignment used in the method `prison_specific_id_requirements` we can see that it is used in the `app/views/content/_id_requirements.text.erb` and `app/views/content/_id_requirements.html.erb` & what a surprise is being passed the sessions prison name via the prisoner in order for lookup in the prison hash.

And then for the actual contents of this method would it make more sense for a visit to be asked what prison is assigned to it and return said object or `nil`?

That way we could do something like..

```ruby
# please rename me to prison :sob:

def prison_data
  visit.prisoner.prison
end

# which could be refactored eventually to something like..

delegate :prison, to: :visit
```